src/pages/VaccineModulePage.js

```diff
@@ -0,0 +1,264 @@
+    const { minTemperature, maxTemperature } = fridge.getTemperatureExposure();
+
+    const { greyTextStyleSmall, greyTextStyleLarge, fridgeInfoSectionStyle } = localStyles;
+    const extraSectionStyle = { flexDirection: 'row', justifySelf: 'flex-end', marginRight: 10 };
+
+    return (
+      <View style={[fridgeInfoSectionStyle, extraSectionStyle]}>
+        {hasBreaches && (
+          <View style={[fridgeInfoSectionStyle, { marginRight: 10 }]}>
+            <Text style={greyTextStyleSmall}>Breaches:</Text>
+            <Text style={greyTextStyleLarge}>{numberOfBreaches}</Text>
+          </View>
+        )}
+        <Text style={greyTextStyleSmall}>Exposure:</Text>
+        {this.renderTemperature(20, minTemperature)}
+        <Text style={greyTextStyleSmall}>to</Text>
+        {this.renderTemperature(20, maxTemperature)}
+      </View>
+    );
+  };
+
+  /* Render 'Breach: {num} Exposure: {fromTemp} to {toTemp} */
+  renderFridgeStock = fridge => {
+    const fridgeStock = fridge.getTotalStock();
+    const hasStock = fridgeStock > 0;
+    const { navigateTo } = this.props;
+
+    const onPress = () =>
+      navigateTo('manageVaccineItems', 'Manage Vaccine Items', {
+        initialLocation: fridge,
+      });
+
+    const { fridgeInfoSectionStyle, greyTextStyleLarge, greyTextStyleSmall } = localStyles;
+    const fontStyleLarge = { ...greyTextStyleLarge, color: SUSSOL_ORANGE };
+    const fontStyleSmall = { ...greyTextStyleSmall, color: SUSSOL_ORANGE };
+
+    if (!hasStock) return <Text style={fontStyleSmall}>Empty Fridge</Text>;
+    return (
+      <TouchableOpacity style={fridgeInfoSectionStyle} onPress={onPress}>
+        <Text style={fontStyleSmall}>Total Stock:</Text>
+        <Text style={fontStyleLarge}>{fridgeStock}</Text>
+        {this.renderIcon('angle-double-right', CHERVON_ICON_STYLE)}
+      </TouchableOpacity>
+    );
+  };
+
+  /* Fridge and all of it's components */
+  renderFridge = fridge => {
+    const { sectionStyle, fridgeInfoSectionStyle } = localStyles;
+    const { selectedFridgeCode } = this.state;
+    const isFridgeSelected = fridge.code === selectedFridgeCode;
+
+    return (
+      <View key={fridge.code}>
+        <View style={[sectionStyle, { flexDirection: 'column', alignItems: 'stretch' }]}>
+          <View style={fridgeInfoSectionStyle}>
+            {this.renderFridgeName(fridge, isFridgeSelected)}
+            {this.renderTemperature(30, fridge.getCurrentTemperature())}
+
+            {fridge.isInBreach ? this.renderIcon('warning', BREACH_ICON_STYLE) : null}
+            <View style={[fridgeInfoSectionStyle, { justifyContent: 'flex-end', flexGrow: 1 }]}>
+              {this.renderFridgeExtraInfo(fridge)}
+              {this.renderFridgeStock(fridge)}
+            </View>
+          </View>
+          {isFridgeSelected && (
+            <View style={{ height: 250, alignSelf: 'stretch' }}>
+              <VaccineChart {...fridge.getTemperaturePoints()} />
```

Comment:
with the above, I this becomes `getAggregatedLogData({sensorLogs: fridge.sensorLogs, minOrMaxOrBoth: 'both', hazards: true, fromDate: new Date(), toDate: new Date()})` - (not exactly but maybe you get it?) - or, `Location.getTempPoints` is just that call, which calculates the `toDate` as like.. -6 months from the `fromDate = new Date()` ?

Reply:
i think i've deal with this in refinement branch, let me know if it's not so